kiva/fonttools/font.py

```diff
@@ -14,62 +14,157 @@
+    'typewriter': TELETYPE,
+    'teletype': TELETYPE,
 }
-font_styles = {"italic": ITALIC}
-font_weights = {"bold": WEIGHT_BOLD}
-font_noise = {"pt", "point", "family"}
+WEIGHTS = {
+    'thin': WEIGHT_THIN,
+    'extra-light': WEIGHT_EXTRALIGHT,
+    'light': WEIGHT_LIGHT,
+    'regular': WEIGHT_NORMAL,
+    'medium': WEIGHT_MEDIUM,
+    'demi-bold': WEIGHT_SEMIBOLD,
+    'bold': WEIGHT_BOLD,
+    'extra-bold': WEIGHT_EXTRABOLD,
+    'heavy': WEIGHT_HEAVY,
+    'extra-heavy': WEIGHT_EXTRAHEAVY
+}
+STYLES = {
+    'italic': ITALIC,
+    'oblique': ITALIC,
+}
+DECORATIONS = {'underline'}
+NOISE = {'pt', 'point', 'px', 'family'}
+
+
+class FontParseError(ValueError):
+    """An exception raised when font parsing fails."""
+    pass
+
+
+def simple_parser(description):
+    """An extremely simple font description parser.
+
+    The parser is simple, and works by splitting the description on whitespace
+    and examining each resulting token for understood terms:
+
+    Size
+        The first numeric term is treated as the font size.
+
+    Weight
+        The following weight terms are accepted: 'thin', 'extra-light',
+        'light', 'regular', 'medium', 'demi-bold', 'bold', 'extra-bold',
+        'heavy', 'extra-heavy'.
+
+    Style
+        The following style terms are accepted: 'italic', 'oblique'.
+
+    Decorations
+        The following decoration terms is accepted: 'underline'
+
+    Generic Families
+        The following generic family terms are accepted: 'default', 'fantasy',
+        'decorative', 'serif', 'roman', 'cursive', 'script', 'sans-serif',
+        'swiss', 'monospace', 'modern', 'typewriter', 'teletype'.
+
+    In addtion, the parser ignores the terms 'pt', 'point', 'px', and 'family'.
+    Any remaining terms are combined into the typeface name.  There is no
+    expected order to the terms.
+
+    This parser is roughly compatible with the various ad-hoc parsers in
+    TraitsUI and Kiva, allowing for the slight differences between them and
+    adding support for additional options supported by Pyface fonts, such as
+    stretch and variants.
 
+    Parameters
+    ----------
+    description : str
+        The font description to be parsed.
 
-def str_to_font(fontspec):
+    Returns
+    -------
+    properties : dict
+        Font properties suitable for use in creating a Pyface Font.
+
+    Notes
+    -----
+    This is not a particularly good parser, as it will fail to properly
+    parse something like "10 pt times new roman" or "14 pt computer modern"
+    since they have generic font names as part of the font face name.
+
+    This is derived from Pyface's equivalent simple_parser.  Eventually both
+    will be replaced by better parsers that can parse something closer to a
+    CSS font definition.
+    """
+    face = []
+    family = DEFAULT
+    size = None
+    weight = WEIGHT_NORMAL
+    style = NORMAL
+    underline = False
+    for word in description.split():
+        lower_word = word.lower()
```

Comment:
Nitpick: for case normalization, `str.casefold` is preferred over `str.lower` (it has better properties with respect to non-Roman character sets). It probably doesn't matter much in this instance.

Reply:
It may - fonts for non-Roman languages may have names that are not in Roman character sets.

There may be other places where this is an issue.

Reply:
```suggestion
        lower_word = word.casefold()
```